north/src/runtime/linux/mod.rs

```diff
@@ -43,5 +43,52 @@ pub async fn init(config: &Config) -> Result<()> {
     debug!("Entering mount namespace");
     sched::unshare(sched::CloneFlags::CLONE_NEWNS).context("Failed to enter mount namespace")?;
 
+    // Pipe minijail log to the runtime
+    capture_minijail_log().await?;
+
+    Ok(())
+}
+
+async fn capture_minijail_log() -> Result<()> {
+    use async_std::{os::unix::io::FromRawFd, stream::StreamExt};
+    use futures::io::AsyncBufReadExt;
+
+    #[allow(non_camel_case_types)]
+    #[allow(dead_code)]
+    #[repr(i32)]
+    enum SyslogLevel {
+        LOG_EMERG = 0,
+        LOG_ALERT = 1,
+        LOG_CRIT = 2,
+        LOG_ERR = 3,
+        LOG_WARNING = 4,
+        LOG_NOTICE = 5,
+        LOG_INFO = 6,
+        LOG_DEBUG = 7,
+        MAX = i32::MAX,
+    }
+
+    if let Some(log_level) = log::max_level().to_level() {
+        let minijail_log_level = match log_level {
+            Level::Error => SyslogLevel::LOG_ERR,
+            Level::Warn => SyslogLevel::LOG_WARNING,
+            Level::Info => SyslogLevel::LOG_INFO,
+            Level::Debug => SyslogLevel::LOG_DEBUG,
+            Level::Trace => SyslogLevel::MAX,
+        };
+
+        let (readfd, writefd) = nix::unistd::pipe()?;
```

Comment:
How it the pipe remove upon process exit?

Reply:
I don't seem to find any specifics, it would probably be cleaned up without much problem. What seems to be important is that if the reading part is closed, the writing part will emit a `SIGPIPE` that could kill the writer, but in this case we are not closing either.

Reply:
Are pipes removed when the process exists that created the pipe? Wouldn't make much sense.

Reply:
well, yes, they live in memory.

Reply:
Yes, linux will close any file descriptor created by the exiting process and that should include this. This should be fine.